lib/kennel/models/dashboard.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+module Kennel
+  module Models
+    class Dashboard < Base
+      include TemplateVariables
+      include OptionalValidations
+
+      API_LIST_INCOMPLETE = true
+      WIDGET_DEFAULTS = {
+        time: {},
+        timeframe: "1h"
+      }.freeze
+      SCREEN_DEFAULTS = { template_variables: [] }.freeze
+      WIDGET_READONLY = [:id].freeze
+      READONLY_ATTRIBUTES = Base::READONLY_ATTRIBUTES + [:author_handle, :modified_at, :url]
+
+      settings :id, :title, :description, :widgets, :kennel_id
+
+      defaults(
+        description: -> { "" },
+        widgets: -> { [] },
+        id: -> { nil }
+      )
+
+      attr_reader :project
+
+      def initialize(project, *args)
+        @project = project
+        super(*args)
+      end
+
+      def self.api_resource
+        "dashboard"
+      end
+
+      def as_json
+        return @json if @json
+        @json = {
+          title: "#{title}#{LOCK}",
+          description: description,
+          layout_type: "ordered",
+          is_read_only: false,
```

Comment:
add to readonly attributes, same as layout_type + notify_list so we never modify it ?